asset/records.go

```diff
@@ -31,6 +26,15 @@ const (
 	LeafScriptVersion       LeafTlvType = 8
 	LeafScriptKey           LeafTlvType = 9
 	LeafGroupKey            LeafTlvType = 10
+
+	// Types for future asset format.
+	// TODO: LeafID and LeafGenesis have conflicting types. The BIP
+	// specifies LeafID, but this would require wallets to store the
+	// preimage to the asset ID elsewhere, when it could be included in the
+	// TLV itself as done with LeafGenesis.
+	LeafAssetID       LeafTlvType = 2
+	LeafAssetType     LeafTlvType = 4
+	LeafXOnlyGroupKey LeafTlvType = 17
```

Comment:
17? 

Why also why would ID and type need to be serialized? You only need the asset ID. You already have the genesis after initial boostrap, so from that you can verify that the asset ID here is valid.